### PR TITLE
[luci-interpreter] Add support for pow op

### PR DESCRIPTION
**Pull request by d-krylov**
Add support for pow  op in luci-interpreter.

ONE-DCO-1.0-Signed-off-by: Denis Krylov <krylovdenism@gmail.com>

### CODE REVIEW

File: compiler/luci-interpreter/src/kernels/Pow.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+Pow::Pow(const Tensor *input1, const Tensor *input2, Tensor *output)
+    : Kernel({input1, input2}, {output})
+{
+}
+
+void Pow::configure()
+{
+  if (input1()->shape() != input2()->shape())
+  {
+    throw std::runtime_error("Input Tensor Shape Mismatch.");
```
Comment (ai-moiseev) [inline compiler/luci-interpreter/src/kernels/Pow.cpp:39]: There's LUCI_INTERPRETER_CHECK macro for this, it's better to use it here, I suppose. But it's nice to have human-readable messages in exceptions.

File: compiler/luci-interpreter/src/kernels/Pow.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  }
+
+  const Shape &output_shape = input1()->shape();
+
+  output()->resize(output_shape);
+}
+
+void Pow::execute() const
+{
+  tflite::reference_ops::Pow(getTensorShape(input1()), getTensorData<float>(input1()),
```
Comment (struss) [inline compiler/luci-interpreter/src/kernels/Pow.cpp:54]: This is only for `float` case and non-broadcast case only. I think this operation need to update for `uint8` case and broadcast case.

File: compiler/luci-interpreter/src/kernels/Pow.cpp [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+{
+
+Pow::Pow(const Tensor *input1, const Tensor *input2, Tensor *output)
+    : Kernel({input1, input2}, {output})
+{
+}
+
+void Pow::configure()
+{
+  if (input1()->shape() != input2()->shape())
```
Comment (binarman) [inline compiler/luci-interpreter/src/kernels/Pow.cpp:37]: TFLite support broadcast, I think we need to check this case as well. See Add operator for broadcast example.

File: compiler/luci-interpreter/src/kernels/Pow.cpp [hunk 4]
```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ * Copyright 2019 The TensorFlow Authors. All Rights Reserved.
```
Comment (s-barannikov) [inline compiler/luci-interpreter/src/kernels/Pow.cpp:3]: Does not look necessary

File: compiler/luci-interpreter/src/kernels/Pow.test.cpp [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+  kernel.configure();
+  kernel.execute();
+
+  EXPECT_THAT(extractTensorData<int32_t>(output_tensor), IntArrayEq(test_outputs));
+  EXPECT_THAT(extractTensorShape(output_tensor), ::testing::ElementsAreArray(base_shape));
+}
+
+TEST(PowTest, Input_Output_Type_NEG)
+{
+  std::initializer_list<int32_t> base_shape = {1, 1};
```
Comment (s-barannikov) [inline compiler/luci-interpreter/src/kernels/Pow.test.cpp:91]: unused

File: compiler/luci-interpreter/src/kernels/TestUtils.cpp [hunk 6]
```diff
@@ -51,6 +51,17 @@ Matcher<std::vector<float>> FloatArrayNear(const std::vector<float> &values, flo
   return ElementsAreArray(matchers);
 }
 
+Matcher<std::vector<int32_t>> IntArrayEq(const std::vector<int32_t> &values)
```
Comment (s-barannikov) [inline compiler/luci-interpreter/src/kernels/TestUtils.cpp:54]: Seems that `::testing::ElementsAreArray` does exactly the same, why don't use it?

**Review by s-barannikov — APPROVED** [review]

**Review by binarman — APPROVED** [review]
LGTM

P.s. We can add more checks (for example in case of non broadcasted shapes)
and add several negative tests for incompatible shapes and unsupported types

**Review by a-zh — APPROVED** [review]
LGTM

**Comment by struss** [comment]
LGTM

**Review by struss — APPROVED** [review]
LGTM

**Review by ai-moiseev — APPROVED** [review]
LGTM

**Review by Hyunjun85 — APPROVED** [review]

**Pull request merged**